### ISSUE TITLE
Example not working: AttributeError: module 'tensorflow' has no attribute 'data'

### ISSUE THREAD

**Issue by renepickhardt** [jfkirk/tensorrec#70]
For reproducability: I have a clean Anaconda installation and successfully installed tensorflow and tensorrec in the following way via python3.6 in jupyter:
``` 
!conda install -y -c conda-forge tensorflow 

olving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.4.10
  latest version: 4.5.4

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /Users/rpickhardt/anaconda3

  added / updated specs: 
    - tensorflow


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    mock-2.0.0                 |           py36_0         103 KB  conda-forge
    protobuf-3.5.2             |           py36_0         603 KB  conda-forge
    certifi-2018.1.18          |           py36_0         143 KB  conda-forge
    tensorflow-1.1.0           |           py36_0        24.8 MB  conda-forge
    libprotobuf-3.5.2          |                0         3.9 MB  conda-forge
    pbr-4.0.4                  |             py_0          61 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        29.7 MB

The following NEW packages will be INSTALLED:

    libprotobuf: 3.5.2-0          conda-forge
    mock:        2.0.0-py36_0     conda-forge
    pbr:         4.0.4-py_0       conda-forge
    protobuf:    3.5.2-py36_0     conda-forge
    tensorflow:  1.1.0-py36_0     conda-forge

The following packages will be UPDATED:

    certifi:     2018.1.18-py36_0             --> 2018.1.18-py36_0 conda-forge


Downloading and Extracting Packages
mock 2.0.0: ############################################################ | 100% 
protobuf 3.5.2: ######################################################## | 100% 
certifi 2018.1.18: ##################################################### | 100% 
tensorflow 1.1.0: ###################################################### | 100% 
libprotobuf 3.5.2: ##################################################### | 100% 
pbr 4.0.4: ############################################################# | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

next I did 
```
!pip install tensorrec

Collecting tensorrec
  Downloading https://files.pythonhosted.org/packages/a3/f3/ba53a0c6112df91dbe3cdbf5b008269b3e518afb43d29160341e5a883231/tensorrec-0.25.2.tar.gz
Building wheels for collected packages: tensorrec
  Running setup.py bdist_wheel for tensorrec ... done
  Stored in directory: /Users/rpickhardt/Library/Caches/pip/wheels/40/80/bc/97d2d1f08db563cde3166bf322d9e9433f160bf424bbd31773
Successfully built tensorrec
Installing collected packages: tensorrec
Successfully installed tensorrec-0.25.2
You are using pip version 9.0.1, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
Both seemed to have worked fine. however running the example from git I get the following stack trace: 

```
Generating positive interactions
Generating negative interactions
Generating user features
Generating item features
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-4d84d711fe05> in <module>()
     13 
     14 # Fit the model for 5 epochs
---> 15 model.fit(interactions, user_features, item_features, epochs=5, verbose=True)
     16 
     17 # Predict scores and ranks for all users and all items

~/anaconda3/lib/python3.6/site-packages/tensorrec/tensorrec.py in fit(self, interactions, user_features, item_features, epochs, learning_rate, alpha, verbose, user_batch_size, n_sampled_items)
    531                          verbose=verbose,
    532                          user_batch_size=user_batch_size,
--> 533                          n_sampled_items=n_sampled_items)
    534 
    535     def fit_partial(self, interactions, user_features, item_features, epochs=1, learning_rate=0.1,

~/anaconda3/lib/python3.6/site-packages/tensorrec/tensorrec.py in fit_partial(self, interactions, user_features, item_features, epochs, learning_rate, alpha, verbose, user_batch_size, n_sampled_items)
    578         # Check if the iterators have been constructed. If not, build them.
    579         if self.tf_interaction_iterator is None:
--> 580             self._build_input_iterators()
    581 
    582         if verbose:

~/anaconda3/lib/python3.6/site-packages/tensorrec/tensorrec.py in _build_input_iterators(self)
    260 
    261     def _build_input_iterators(self):
--> 262         self.tf_user_feature_iterator = create_tensorrec_iterator(name='tf_user_feature_iterator')
    263         self.tf_item_feature_iterator = create_tensorrec_iterator(name='tf_item_feature_iterator')
    264         self.tf_interaction_iterator = create_tensorrec_iterator(name='tf_interaction_iterator')

~/anaconda3/lib/python3.6/site-packages/tensorrec/input_utils.py in create_tensorrec_iterator(name)
     13     :return: tf.data.Iterator
     14     """
---> 15     return tf.data.Iterator.from_structure(
     16             output_types=(tf.int64, tf.int64, tf.float32, tf.int64, tf.int64),
     17             output_shapes=([None], [None], [None], [], []),

AttributeError: module 'tensorflow' has no attribute 'data'
```
This is strange because according to the tensorflow doc this should work (c.f.: https://www.tensorflow.org/api_docs/python/tf/data/Iterator ) However I double checked at least the python object does indeed not have the field:

```
import tensorflow as tf
for k in tf.__dict__.keys():
    print(k)

__name__
__doc__
__package__
__loader__
__spec__
__path__
__file__
__cached__
__builtins__
AggregationMethod
Assert
AttrValue
COMPILER_VERSION
ConditionalAccumulator
ConditionalAccumulatorBase
ConfigProto
DType
DeviceSpec
Dimension
Event
FIFOQueue
FixedLenFeature
FixedLenSequenceFeature
FixedLengthRecordReader
GIT_VERSION
GPUOptions
GRAPH_DEF_VERSION
GRAPH_DEF_VERSION_MIN_CONSUMER
GRAPH_DEF_VERSION_MIN_PRODUCER
Graph
GraphDef
GraphKeys
GraphOptions
HistogramProto
IdentityReader
IndexedSlices
InteractiveSession
LogMessage
NameAttrList
NoGradient
NodeDef
NotDifferentiable
OpError
Operation
OptimizerOptions
PaddingFIFOQueue
Print
PriorityQueue
QUANTIZED_DTYPES
QueueBase
RandomShuffleQueue
ReaderBase
RegisterGradient
RunMetadata
RunOptions
Session
SessionLog
SparseConditionalAccumulator
SparseFeature
SparseTensor
SparseTensorValue
Summary
TFRecordReader
Tensor
TensorArray
TensorInfo
TensorShape
TextLineReader
VERSION
VarLenFeature
Variable
VariableScope
WholeFileReader
__compiler_version__
__git_version__
__version__
abs
accumulate_n
acos
add
add_check_numerics_ops
add_n
add_to_collection
all_variables
app
arg_max
arg_min
argmax
argmin
as_dtype
as_string
asin
assert_equal
assert_greater
assert_greater_equal
assert_integer
assert_less
assert_less_equal
assert_negative
assert_non_negative
assert_non_positive
assert_none_equal
assert_positive
assert_proper_iterable
assert_rank
assert_rank_at_least
assert_type
assert_variables_initialized
assign
assign_add
assign_sub
atan
batch_to_space
batch_to_space_nd
betainc
bfloat16
bincount
bitcast
bool
boolean_mask
broadcast_dynamic_shape
broadcast_static_shape
case
cast
ceil
check_numerics
cholesky
cholesky_solve
clip_by_average_norm
clip_by_global_norm
clip_by_norm
clip_by_value
compat
complex
complex128
complex64
concat
cond
confusion_matrix
conj
constant
constant_initializer
container
control_dependencies
convert_to_tensor
convert_to_tensor_or_indexed_slices
convert_to_tensor_or_sparse_tensor
cos
count_nonzero
count_up_to
create_partitioned_variables
cross
cumprod
cumsum
decode_base64
decode_csv
decode_json_example
decode_raw
delete_session_tensor
depth_to_space
dequantize
deserialize_many_sparse
device
diag
diag_part
digamma
div
divide
double
dynamic_partition
dynamic_stitch
edit_distance
einsum
encode_base64
equal
erf
erfc
errors
estimator
exp
expand_dims
expm1
extract_image_patches
eye
fake_quant_with_min_max_args
fake_quant_with_min_max_args_gradient
fake_quant_with_min_max_vars
fake_quant_with_min_max_vars_gradient
fake_quant_with_min_max_vars_per_channel
fake_quant_with_min_max_vars_per_channel_gradient
fft
fft2d
fft3d
fill
fixed_size_partitioner
flags
float16
float32
float64
floor
floor_div
floordiv
floormod
foldl
foldr
gather
gather_nd
get_collection
get_collection_ref
get_default_graph
get_default_session
get_local_variable
get_seed
get_session_handle
get_session_tensor
get_variable
get_variable_scope
gfile
global_norm
global_variables
global_variables_initializer
gradients
graph_util
greater
greater_equal
group
half
hessians
histogram_fixed_width
identity
ifft
ifft2d
ifft3d
igamma
igammac
imag
image
import_graph_def
initialize_all_tables
initialize_all_variables
initialize_local_variables
initialize_variables
int16
int32
int64
int8
invert_permutation
is_finite
is_inf
is_nan
is_non_decreasing
is_numeric_tensor
is_strictly_increasing
is_variable_initialized
layers
lbeta
less
less_equal
lgamma
lin_space
linspace
load_file_system_library
load_op_library
local_variables
local_variables_initializer
log
log1p
logging
logical_and
logical_not
logical_or
logical_xor
losses
make_template
map_fn
matching_files
matmul
matrix_band_part
matrix_determinant
matrix_diag
matrix_diag_part
matrix_inverse
matrix_set_diag
matrix_solve
matrix_solve_ls
matrix_transpose
matrix_triangular_solve
maximum
meshgrid
metrics
min_max_variable_partitioner
minimum
mod
model_variables
moving_average_variables
multinomial
multiply
name_scope
negative
newaxis
nn
no_op
no_regularizer
norm
not_equal
one_hot
ones
ones_initializer
ones_like
op_scope
orthogonal_initializer
pad
parallel_stack
parse_example
parse_single_example
parse_single_sequence_example
parse_tensor
placeholder
placeholder_with_default
polygamma
pow
py_func
python_io
pywrap_tensorflow
qint16
qint32
qint8
qr
quantize_v2
quantized_concat
quint16
quint8
random_crop
random_gamma
random_normal
random_normal_initializer
random_poisson
random_shuffle
random_uniform
random_uniform_initializer
range
rank
read_file
real
realdiv
reciprocal
reduce_all
reduce_any
reduce_join
reduce_logsumexp
reduce_max
reduce_mean
reduce_min
reduce_prod
reduce_sum
register_tensor_conversion_function
report_uninitialized_variables
required_space_to_batch_paddings
reset_default_graph
reshape
resource
resource_loader
reverse
reverse_sequence
reverse_v2
rint
round
rsqrt
saturate_cast
saved_model
scalar_mul
scan
scatter_add
scatter_div
scatter_mul
scatter_nd
scatter_nd_add
scatter_nd_sub
scatter_nd_update
scatter_sub
scatter_update
sdca
segment_max
segment_mean
segment_min
segment_prod
segment_sum
self_adjoint_eig
self_adjoint_eigvals
sequence_mask
serialize_many_sparse
serialize_sparse
set_random_seed
setdiff1d
sets
shape
shape_n
sigmoid
sign
sin
size
slice
space_to_batch
space_to_batch_nd
space_to_depth
sparse_add
sparse_concat
sparse_fill_empty_rows
sparse_mask
sparse_matmul
sparse_maximum
sparse_merge
sparse_minimum
sparse_placeholder
sparse_reduce_sum
sparse_reduce_sum_sparse
sparse_reorder
sparse_reset_shape
sparse_reshape
sparse_retain
sparse_segment_mean
sparse_segment_sqrt_n
sparse_segment_sum
sparse_softmax
sparse_split
sparse_tensor_dense_matmul
sparse_tensor_to_dense
sparse_to_dense
sparse_to_indicator
sparse_transpose
spectral
split
sqrt
square
squared_difference
squeeze
stack
stop_gradient
strided_slice
string
string_join
string_split
string_to_hash_bucket
string_to_hash_bucket_fast
string_to_hash_bucket_strong
string_to_number
substr
subtract
summary
svd
sysconfig
tables_initializer
tan
tanh
tensordot
test
tile
to_bfloat16
to_double
to_float
to_int32
to_int64
trace
train
trainable_variables
transpose
truediv
truncated_normal
truncated_normal_initializer
truncatediv
truncatemod
tuple
uint16
uint8
uniform_unit_scaling_initializer
unique
unique_with_counts
unsorted_segment_max
unsorted_segment_sum
unstack
user_ops
variable_axis_size_partitioner
variable_op_scope
variable_scope
variables_initializer
verify_tensor_all_finite
where
while_loop
write_file
zeros
zeros_initializer
zeros_like
zeta
_LazyContribLoader
contrib
```

in [tensorflow issue tracker in issue 2879](https://github.com/tensorflow/models/issues/2879) one finds that tf.data has been introduced in tf 1.4 and was named tf.contrib.data before so apparently my tensorflow installation is too old. 

indeed prints the same as anaconda installation. 
```
print(tf.__version__)
1.1.0
```
I will attach a patch of your software in a sec so that it will checked that the proper TensorFlow version is installed. 

as a side note upgrading to version 1.4 is not enough because then we get: AttributeError: 'Iterator' object has no attribute '_output_classes' as in #66  where it is stated that tensorflow version 1.7 is mandatory. 
actually using tensorflow 1.8. also seems to work just fine. 

For anaconda users the following line will fix the issue (for some reason tf 1.7 is not within the conda-froge): 
```
!conda install -y -c conda-forge tensorflow=1.8.0
```

**Comment by jfkirk** [comment]
@renepickhardt We've merged your PR and I've done a patch release (0.25.3) that includes your version checking. Do you think there is anything outstanding for this issue, or is it ready to close?

**Comment by renepickhardt** [comment]
I have just closed it. I made it so extensive because I think hackers google stuff and if people run into problems this could be helpful. but google should also display closed issues (:

the only thing that could also be included is the dependencies of tensorflow in the readme. but I guess with version checking now everything should be fine
